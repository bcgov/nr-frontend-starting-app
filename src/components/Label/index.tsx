import React from 'react';

interface LabelProps {
  labelStr: string,
  forStr: string
}

const Label = ({ labelStr, forStr }: LabelProps) => (
  <label className="control-label" htmlFor={forStr}>{ labelStr }</label>
);

export default Label;
