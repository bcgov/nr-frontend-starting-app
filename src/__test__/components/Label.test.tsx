/* eslint-disable no-undef */
import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';
import Label from '../../components/Label';
import '@testing-library/jest-dom';

describe('the Label component', () => {
  it('should have the correct text', () => {
    const { getByTestId } = render(<Label labelStr="Test label" forStr="" />);
    expect(getByTestId('label-element').textContent).toBe('Test label');
  });

  it('should have the correct "for" attribute', () => {
    const { getByTestId } = render(<Label labelStr="Test label" forStr="testInput" />);
    expect(getByTestId('label-element')).toHaveAttribute('for', 'testInput');
  });

  it('should match the snapshot', () => {
    const tree = renderer
      .create(<Label labelStr="Test label" forStr="" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
