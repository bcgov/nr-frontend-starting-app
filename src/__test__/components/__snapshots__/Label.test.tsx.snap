// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the component renders correctly should match the snapshot 1`] = `
<label
  className="control-label"
  data-testid="label-element"
  htmlFor=""
>
  Test label
</label>
`;
